rasa/nlu/components.py

```diff
@@ -179,28 +181,102 @@ def validate_pipeline(pipeline: List["Component"]) -> None:
+
+
 def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ) -> None:
     """Validates that all components are present in the pipeline based on data.
 
     Args:
-        pipeline: The list of the :class:`rasa.nlu.components.Component`.
+        pipeline: The list of the :class:`rasa.nlu.components.Component`s.
         data: The :class:`rasa.nlu.training_data.training_data.TrainingData`.
     """
 
-    from rasa.nlu.selectors.response_selector import ResponseSelector
+    if data.response_examples and not any_components_in_pipeline(
+        ["ResponseSelector"], pipeline
+    ):
+        raise_warning(
+            "You have defined training data with examples for training a response selector, but "
+            "your NLU pipeline does not include a response selector component. To train a "
+            "model on your response selector data, add a 'ResponseSelector' to your pipeline."
+        )
+
+    if data.entity_examples and not any_components_in_pipeline(
+        TRAINABLE_EXTRACTORS, pipeline
+    ):
+        raise_warning(
+            "You have defined training data consisting of entity examples, but "
+            "your NLU pipeline does not include an entity extractor trained on "
+            "your training data. To extract non-pretrained entities, add one of "
+            f"{TRAINABLE_EXTRACTORS} to your pipeline."
+        )
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if isinstance(component, ResponseSelector):
-            response_selector_exists = True
+    if data.regex_features and not any_components_in_pipeline(
+        ["RegexFeaturizer"], pipeline
+    ):
+        raise_warning(
+            "You have defined training data with regexes, but "
+            "your NLU pipeline does not include a 'RegexFeaturizer'. "
+            "To featurize regexes, include a 'RegexFeaturizer' in your pipeline."
+        )
 
-    if len(data.response_examples) and not response_selector_exists:
+    if data.lookup_tables and not any_components_in_pipeline(
+        ["RegexFeaturizer"], pipeline
+    ):
+        raise_warning(
+            "You have defined training data consisting of lookup tables, but "
+            "your NLU pipeline does not include a 'RegexFeaturizer'. "
+            "To featurize lookup tables, add a 'RegexFeaturizer' to your pipeline."
+        )
+
+    if data.lookup_tables:
+        if not any_components_in_pipeline(
+            ["CRFEntityExtractor", "DIETClassifier"], pipeline
+        ):
+            raise_warning(
+                "You have defined training data consisting of lookup tables, but "
+                "your NLU pipeline does not include any components that use these features. "
+                "To featurize lookup tables, add a 'DIETClassifier' or a 'CRFEntityExtractor' with "
```

Comment:
```suggestion
                "To make use of lookup tables, add a 'DIETClassifier' or a 'CRFEntityExtractor' with "
```
We are not featurizing lookup tables in `DIETClassifier`. We only do that in `CRFEntityExtractor`.

Reply:
Got it, will fix. How are they used in DIET then?

Reply:
`RegexFeaturizer` adds them as sparse features. `DIETClassifier` simply uses the sparse features.